app/src/main/java/org/schabi/newpipe/settings/custom/NotificationSlot.java

```diff
@@ -0,0 +1,172 @@
+
+import org.schabi.newpipe.R;
+import org.schabi.newpipe.databinding.ListRadioIconItemBinding;
+import org.schabi.newpipe.databinding.SingleChoiceDialogViewBinding;
+import org.schabi.newpipe.player.notification.NotificationConstants;
+import org.schabi.newpipe.util.DeviceUtils;
+import org.schabi.newpipe.util.ThemeHelper;
+import org.schabi.newpipe.views.FocusOverlayView;
+
+import java.util.Objects;
+import java.util.function.BiConsumer;
+
+class NotificationSlot {
+
+    private static final int[] SLOT_ITEMS = {
+            R.id.notificationAction0,
+            R.id.notificationAction1,
+            R.id.notificationAction2,
+            R.id.notificationAction3,
+            R.id.notificationAction4,
+    };
+
+    private static final int[] SLOT_TITLES = {
+            R.string.notification_action_0_title,
+            R.string.notification_action_1_title,
+            R.string.notification_action_2_title,
+            R.string.notification_action_3_title,
+            R.string.notification_action_4_title,
+    };
+
+    private final int i;
+    private @NotificationConstants.Action int selectedAction;
+    private final Context context;
+    private final BiConsumer<Integer, CheckBox> onToggleCompactSlot;
+
+    private ImageView icon;
+    private TextView summary;
+
+    NotificationSlot(final Context context,
+                     final SharedPreferences prefs,
+                     final int actionIndex,
+                     final View parentView,
+                     final boolean isCompactSlotChecked,
+                     final BiConsumer<Integer, CheckBox> onToggleCompactSlot) {
+        this.context = context;
+        this.i = actionIndex;
+        this.onToggleCompactSlot = onToggleCompactSlot;
+
+        selectedAction = Objects.requireNonNull(prefs).getInt(
+                context.getString(NotificationConstants.SLOT_PREF_KEYS[i]),
+                NotificationConstants.SLOT_DEFAULTS[i]);
+        final View view = parentView.findViewById(SLOT_ITEMS[i]);
+
+        // only show the last two notification slots on Android 13+
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU || i >= 3) {
+            setupSelectedAction(view);
+            setupTitle(view);
+            setupCheckbox(view, isCompactSlotChecked);
+        } else {
+            view.setVisibility(View.GONE);
+        }
+    }
+
+    void setupTitle(final View view) {
+        ((TextView) view.findViewById(R.id.notificationActionTitle))
+                .setText(SLOT_TITLES[i]);
+        view.findViewById(R.id.notificationActionClickableArea).setOnClickListener(
+                v -> openActionChooserDialog());
+    }
+
+    void setupCheckbox(final View view, final boolean isCompactSlotChecked) {
+        final CheckBox compactSlotCheckBox = view.findViewById(R.id.notificationActionCheckBox);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            // there are no compact slots to customize on Android 33+
```

Comment:
Either Android 13+ or API 33+
```suggestion
            // there are no compact slots to customize on Android 13+
```